File: malevich-server/src/main/java/io/malevich/server/transfer/ParticipantDto.java
package io.malevich.server.transfer;

import com.yinyang.core.server.transfer.FileDto;
import com.yinyang.core.server.transfer.UserDto;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.Map;

@Getter
@Setter
public class ParticipantDto {

    private Long id;

    private ParticipantTypeDto type;

    private String phoneNumber;

    private CountryDto country;

    private List<UserDto> users;

    private List<AddressDto> addresses;

    private FileDto thumbnail;

    private KycLevelDto kycLevel;

    //gallery
    private OrganizationDto organization;

    private FileDto image;

    private Map<String, String> titleMl;

    private Map<String, String> descriptionMl;

    //trader
    private PersonDto person;

}

File: malevich-web/src/app/_transfer/participantDto.ts
import {CountryDto} from "./countryDto";
import {AddressDto} from "./addressDto";
import {ParticipantTypeDto} from "./participantTypeDto";
import {PersonDto} from "./personDto";
import {OrganizationDto} from "./organizationDto";
import {UserDto, FileDto} from "../../../node_modules/yinyang-core";

export class ParticipantDto {
  id: number;
  type: ParticipantTypeDto;
  users: UserDto[];
  phoneNumber: string;
  country: CountryDto;
  thumbnail: FileDto;
  addresses: AddressDto[];

  organization: OrganizationDto;
  image: FileDto;
  descriptionMl: Map<string, string>;

  person: PersonDto;
}
